src/visitable.cpp

```diff
@@ -121,7 +135,7 @@ bool visitable<T>::has_quality( const quality_id &qual, int level, int qty ) con
 template <>
 bool visitable<inventory>::has_quality( const quality_id &qual, int level, int qty ) const
 {
-    long res = 0;
+    int res = 0;
```

Comment:
Why was it long in the first place?  The variables need to be consistently int or consistently long.